contracts/modules/andromeda-shunting/README.md

```diff
@@ -0,0 +1,152 @@
+
+## InstantiateMsg
+
+```rust
+pub struct InstantiateMsg {
+    pub expressions: Vec<String>,
+    pub kernel_address: String,
+    pub owner: Option<String>
+}
+```
+```json
+{
+    "expressions": ["...", ...],
+    "kernel_address": "...",
+}
+```
+
+The `expressions` field is provided as an array of mathematical expressions. For example:
+
+```
+[
+    "x0 ^ 2",
+    "y0 * x1"
+]
+```
+
+In the expression, `x<n>` means the nth parameter in the params of the query, and `y<n>` means the result of the nth expression. In the above example, when evaluating the expression, the first expression calculate the square of the first parameter passed by the query and the second expression get the first expression's result and multiply it by second parameter. Only the last expression's result is returned as a query result. 
+
+_**Warning:** Expressions can only use the previous expression's result._
+
+
+## ExecuteMsg
+_**Warning:** Only owner or operator can update expressions._
+```rust
+pub enum ExecuteMsg {
+    UpdateExpressions { expressions: Vec<String> },
+}
+```
+```json
+{
+    "update_expressions":  {
+        "expressions": ["...", ...] 
+    },
+}
+```
+
+Expressions can be updated using the `ExecuteMsg`.
+
+## QueryMsg
+```rust
+pub enum QueryMsg {
+    #[returns(ShuntingResponse)]
+    Evaluate { params: Vec<EvaluateParam> },
+}
+```
+```json
+{
+    "evaluate":  {
+        "params": [] 
+    },
+}
+```
+
+We use `Evaulate` msg to evaluate the mathematical expressions. It returns a `ShuntingResponse` message which is structured as follows. 
+
+```rust
+pub struct ShuntingResponse {
+    pub result: String,
+}
+```
+```json
+{
+    "result":  "",
+}
+```
+
+
+The `params` of the `Evaluate` message can be either raw value or references to other smart contracts.
+
+```rust
+pub enum EvaluateParam {
+    Value(String),
+    Reference(EvaluateRefParam),
+}
+```
+- `Value(String)` is used to pass raw(numerical) values.
+- `Reference(EvaluateRefParam)` is used to convey the result of another contract.
+
+For instance, if you wish to pass the values 2 and 3.14 as parameters, the following message can be used:
+
+```rust
+Evaluate {
+    params: vec![
+        EvaluateParam::Value("2".to_string()),
+        EvaluateParam::Value("3.14".to_string()),
+    ]
+}
+```
+```json
+{
+    "evaluate": {
+        "params":[{"value":"2"}, {"value":"3.14"}]
+    }
+}
+```
+
+## EvaluateRefParam
+
+```rust
+pub struct EvaluateRefParam {
+    pub contract: Addr,
+    pub msg: String,
+    pub accessor: String,
+}
+```
+- `contract` is the address of the contract intended for integration.
+- `msg` is a `base64` encoded message slated for dispatch to the contract.
+- `accessor` identifies the field within the result that is intended for use as a parameter.
```

Comment:
Add some documentation to explain how this accessor field works

Reply:
Added documentation about using nested resutl.